lib/typhoeus/adapters/faraday.rb

```diff
@@ -162,8 +162,8 @@ def configure_proxy(req, env)
 
       def configure_timeout(req, env)
         env_req = env[:request]
-        req.options[:timeout_ms] = (env_req[:timeout] * 1000).to_i             if env_req[:timeout]
-        req.options[:connecttimeout_ms] = (env_req[:open_timeout] * 1000).to_i if env_req[:open_timeout]
+        req.options[:timeout_ms] = (env_req[:timeout].to_f * 1000).to_i             if env_req[:timeout]
```

Comment:
To avoid any surprises, such as the one you come across, I think it's better to use `Float` instead of `to_f`. The main difference is`Float` will raise an error if the object cannot be cast to a float where as `.to_f` will do its best.
```ruby
Float("1a2")
# => ArgumentError (invalid value for Float(): "1a2")

"1a2".to_f
#=> 1.0
```

Reply:
Just need to another the above as a test case and that should be it I think.